### PR TITLE
Move/renames to make Ray.jl the repo root repository

### PR DESCRIPTION
**Pull request by omus**
Reorganizes the content of this repo so that Ray.jl is the root package and ray_julia_jll.jl is the sub package. Part of: https://github.com/beacon-biosignals/ray_core_worker_julia_jll.jl/issues/91

***IMPORTANT: PR should NOT be squashed as this can confuse Git history***

### CODE REVIEW

**Review by kleinschmidt — APPROVED** [review]
looks good to me pending green CI.  @omus I'll let you handle the merge once it's ready.

File: docs/fix_doctests.jl [hunk 1]
```diff
@@ -12,7 +12,7 @@ DocMeta.setdocmeta!(Ray, :DocTestSetup, :(using Ray);
 
 if get(ENV, "CI", "false") == "true" || success(`git diff --quiet`)
     if ismissing(get(ENV, "AWS_PROFILE", missing))
-        @warn """You may need to set `ENV["AWS_PROFILE"] = ray_core_worker_julia_jll-ci` in order to successfully run the doctests"""
+        @warn """You may need to set `ENV["AWS_PROFILE"] = Ray-ci` in order to successfully run the doctests"""
```
Comment (omus) [inline docs/fix_doctests.jl:15]: Preemptively renamed this profile.

**Comment by omus** [comment]
I'll try to wrap this up tonight. Steps remaining:

- Apply fixup commits and then force push with lease
- Merge this PR via a merge commit (ensures history is properly tracked)
- Rename this repository to be Ray.jl (may require manually deleting existing empty repo).
- Create Terraform PR to remove original repo
- Clear out old GHA cache files to avoid bad builds

**Comment by omus** [comment]
Had the Julia 1.9 tests fail with:
```
Async waiting on task: Test Failed at /home/runner/work/ray_core_worker_julia_jll.jl/ray_core_worker_julia_jll.jl/test/task.jl:195
  Expression: results == [1, 10]
   Evaluated: Any[10, 1] == [1, 10]

Stacktrace:
 [1] macro expansion
   @ /opt/hostedtoolcache/julia/1.9.3/x64/share/julia/stdlib/v1.9/Test/src/Test.jl:478 [inlined]
 [2] macro expansion
   @ ~/work/ray_core_worker_julia_jll.jl/ray_core_worker_julia_jll.jl/test/task.jl:195 [inlined]
 [3] macro expansion
   @ /opt/hostedtoolcache/julia/1.9.3/x64/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
 [4] top-level scope
   @ ~/work/ray_core_worker_julia_jll.jl/ray_core_worker_julia_jll.jl/test/task.jl:187
```
– https://github.com/beacon-biosignals/ray_core_worker_julia_jll.jl/actions/runs/6102536017/job/16561202496?pr=90
 
Looks like something to look into later

**Pull request merged**